pkg/events/client.go

```diff
@@ -0,0 +1,134 @@
+package events
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/google/go-querystring/query"
+	"github.com/workos/workos-go/v2/pkg/workos_errors"
+
+	"github.com/workos/workos-go/v2/internal/workos"
+	"github.com/workos/workos-go/v2/pkg/common"
+)
+
+// ResponseLimit is the default number of records to limit a response to.
+const ResponseLimit = 10
+
+// Client represents a client that performs Event requests to the WorkOS API.
+type Client struct {
+	// The WorkOS API Key. It can be found in https://dashboard.workos.com/api-keys.
+	APIKey string
+
+	// The http.Client that is used to get Event records from WorkOS.
+	// Defaults to http.Client.
+	HTTPClient *http.Client
+
+	// The endpoint to WorkOS API. Defaults to https://api.workos.com.
+	Endpoint string
+
+	once sync.Once
+}
+
+func (c *Client) init() {
+	if c.HTTPClient == nil {
+		c.HTTPClient = &http.Client{Timeout: 10 * time.Second}
+	}
+
+	if c.Endpoint == "" {
+		c.Endpoint = "https://api.workos.com"
+	}
+}
+
+// Event contains data about a particular Event.
+type Event struct {
+	// The Event's unique identifier.
+	ID string `json:"id"`
+
+	// The type of Event.
+	Event string `json:"event"`
+
+	// The Event's data in raw encoded JSON.
+	Data json.RawMessage `json:"data"`
+
+	// The Event's created at date.
+	CreatedAt string `json:"created_at"`
+}
+
+// GetEventsOpts contains the options to request provisioned Events.
+type GetEventsOpts struct {
+	// Filter to only return Events of particular types.
+	Events []string `url:"events,omitempty"`
+
+	// Maximum number of records to return.
+	Limit int `url:"limit"`
+
+	// Pagination cursor to receive records after a provided Event ID.
+	After string `url:"after,omitempty"`
+
+	// Date range start for stream of Events.
+	RangeStart string `url:"after,omitempty"`
+
+	// Date range end for stream of Events.
+	RangeEnd string `url:"after,omitempty"`
+}
+
+// GetEventsResponse describes the response structure when requesting
+// Events.
+type GetEventsResponse struct {
+	// List of Events.
+	Data []Event `json:"data"`
+
+	// Cursor pagination options.
+	ListMetadata common.ListMetadata `json:"listMetadata"`
+}
+
+// GetEvents gets a list of Events.
+func (c *Client) GetEvents(
+	ctx context.Context,
+	opts GetEventsOpts,
+) (GetEventsResponse, error) {
```

Comment:
I wonder if we should following a pointer to a return value model:
`) (*GetEventsResponse, error) {`

and returning `nil` in the error case, what does the rest of the SDK do?:

```
if err != nil {
	return nil, err
}	
```

Reply:
It seems we actually return an empty response struct instead of nil across the SDK, probably makes sense to continue the pattern (although I'm open to following the pattern ^^ but  we should make it uniform across the SDK

Reply:
FWIW, if it's a good practice - we counld add this to the next major release as an update. I was going to start on some V3 changes on a new branch.